app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala

```diff
@@ -833,32 +912,8 @@ class SectionRenderingService(
               upscanInitiate
             )
           case FileUpload(fileUploadProvider) =>
-            val fileId: FileId =
-              ei.envelope
-                .find(formComponent.modelComponentId)
-                .map(_.fileId)
-                .getOrElse(ei.envelope.mapping.fileIdFor(formComponent.id))
-
+            val fileId: FileId = findFileId(ei, formComponent)
             fileUploadProvider match {
-              case FileUploadProvider.Upscan =>
-                val upscanInitiateResponse: UpscanInitiateResponse = upscanInitiate.get(formComponent.id)
-                val hiddenFields: List[Html] = upscanInitiateResponse.uploadRequest.fields.toList.map {
-                  case (name, value) =>
-                    html.form.snippets.hidden(name, value)
-                }
-                val fileUploadName = "file"
-                val attributes = Map("upscan" -> "upscan")
-                htmlForFileUpload(
-                  formComponent,
-                  formTemplateId,
-                  ei,
-                  validationResult,
-                  fileId,
-                  fileUploadName,
-                  upscanInitiateResponse.uploadRequest.href,
-                  attributes,
-                  hiddenFields
-                )
```

Comment:
We should keep this as it was, we will only need to link it to correct `<form> with htmlId` from before.